### ISSUE TITLE
dollar sign required to inform that field is from an included model.

### ISSUE THREAD

**Issue by carloshmm** [sequelize/sequelize#11365]
I'm trying to search for a field that comes from a included model:
Example:
```
model.findOne({
      where: query,
      include: [
          {
            model: appModel,
            foreignKey: 'appId',
            attributes: ['partnerId']
          }
      ]
});
```
The query value is set to:
`id: mallId, '$app.partner_id$': hash`

I found out about the dollar sign requirement reading these answers from:
https://stackoverflow.com/a/56858118
https://stackoverflow.com/a/56240135

I wasn't able to find associated documentation about this, is it available or not? If not, I think it should be a good idea to add it, I can do a pull request with these instructions if necessary.

**Comment by bradennapier** [comment]
I actually ran into this yesterday and ended up having to comb through stack overflow, old documentation which does mention it, etc.

It is mentioned in the docs but it isn't very clear or prominent.

https://sequelize.org/master/manual/models-usage.html#top-level-where-with-eagerly-loaded-models

**Comment by papb** [comment]
@bradennapier Nice find!! This answers my question from above: it deserves to be documented (more explicitly)

**Comment by papb** [comment]
Closed by #11825 

See the new [guide on eager loading](https://sequelize.org/master/manual/eager-loading.html#complex-where-clauses-at-the-top-level).
